### PR TITLE
Instanced Models

### PR DESCRIPTION
**Pull request by benjxia**
Eliminated the need to re load 3d models from disk every time we want a new instance of some object.

### CODE REVIEW

**Comment by Alanine42** [comment]
works on my machine with this dir structure (pasted from discord):

> client
 |_models
   |_env_fixed.obj
   |_rock_w_tex or whatever

File: common/GameConstants.h [hunk 1]
```diff
@@ -1,5 +1,6 @@
 // Number of entities that can move by itself (4 players for now)
-#define NUM_MOVEMENT_ENTITIES 4
+#define NUM_PLAYER_ENTITIES 4
+#define NUM_MOVEMENT_ENTITIES NUM_PLAYER_ENTITIES + 0
```
Comment (Alanine42) [inline common/GameConstants.h:3]: nice

File: client/src/Client.cpp [hunk 2]
```diff
@@ -17,13 +16,13 @@ int main()
...
-    // todo: We prolly need a separate function to init game entities -- there'll be a lot --- combine with ECS @Matthew
-    for (int i = 0; i < NUM_MOVEMENT_ENTITIES; i++) {
-        // ModelComposite
-        playersModel[i] = new sge::ModelComposite("rock_w_tex/rock2.obj");
+    sge::loadModels();
+
+    // TODO: change this when yall want to
+    entities.push_back(std::make_unique<sge::EntityState>(MAP));
+    for (unsigned int i = 0; i < 4; i++) {
+        entities.push_back(std::make_unique<sge::DynamicEntityState>(PLAYER_0, i));
```
Comment (Alanine42) [inline client/src/Client.cpp:24]: nit: maybe add a comment saying `i = index into every game data array in ClientGame`

**Review by Alanine42 — APPROVED** [review]
lgtm, approved

**Review by Metarus — APPROVED** [review]
works on my machine with the same directory structure. no real functionality is changed (probably performance but that's hard for me to say obviously), but that seems like expected behavior. lgtm

**Comment by benjxia** [comment]
Yea expected behavior is no functionality changed, one issue is that shadows/shading is not reflected on the rock when the user rotates, but that can be fixed when I redo shading.

**Comment by Alanine42** [comment]
6881 musescore

**Pull request merged**